main.go

```diff
@@ -173,10 +173,18 @@ func realMain(ctx context.Context) error {
 
 // buildProxy builds the reverse proxy server, forwarding requests on bind to
 // the provided host.
-func buildProxy(host, bind *url.URL, tokenSource oauth2.TokenSource) *httputil.ReverseProxy {
+// allowInsecureHttps only used to run tests
+func buildProxy(host, bind *url.URL, tokenSource oauth2.TokenSource, enableHttp2 bool, allowInsecureHttps bool) *httputil.ReverseProxy {
 	// Build and configure the proxy.
 	proxy := httputil.NewSingleHostReverseProxy(host)
-
+	// Use http2 for outgoing connections
+	if enableHttp2{
+		proxy.Transport = &http2.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: allowInsecureHttps,
```

Comment:
What is the goal with introducing this configurable?

Reply:
The goal of this is enable skip cert verification when running the tests with http2, is not intended to enable this by any command line flag.

This is needed mostly because on tests you cannot create a plain http2 test server on [net/http/httptest.Server](https://pkg.go.dev/net/http/httptest#Server)

Another approach could be remove this, and override the transport on http2 test to enable skip cert verification on the http2 tests. However, i think this might interfere with the main purpose of the http2 test.

What do you think?

Reply:
What if we used a self-signed cert and overrode the CA bundle?

Reply:
Changed `allowInsecureHttps bool` for `caCertificate *x509.Certificate`